crates/subspace-runtime/src/lib.rs

```diff
@@ -792,26 +813,34 @@ fn extract_call_block_object_mapping(
     objects: &mut Vec<BlockObject>,
     call: &Call,
     recursion_depth_left: u16,
+    successful_calls: &mut VecDeque<Hash>,
 ) {
     // Add enum variant to the base offset.
     base_offset += 1;
 
     match call {
         Call::Feeds(call) => {
-            extract_feeds_block_object_mapping(base_offset, objects, call);
+            extract_feeds_block_object_mapping(base_offset, objects, call, successful_calls);
         }
         Call::ObjectStore(call) => {
             extract_object_store_block_object_mapping(base_offset, objects, call);
         }
         Call::Utility(call) => {
-            extract_utility_block_object_mapping(base_offset, objects, call, recursion_depth_left);
+            extract_utility_block_object_mapping(
+                base_offset,
+                objects,
+                call,
+                recursion_depth_left,
+                successful_calls,
+            );
         }
         _ => {}
     }
 }
 
 fn extract_block_object_mapping(block: Block, successful_calls: Vec<Hash>) -> BlockObjectMapping {
     let mut block_object_mapping = BlockObjectMapping::default();
+    let mut successful_calls = VecDeque::from(successful_calls);
```

Comment:
Why not making it `VecDeque` everywhere in the first place then?

Reply:
Thought about it but didn't want to bring in an implementation detail to the API defintion specifically tied to Feeds

Reply:
I think it is actually fine, we're building API for object extraction. Whether have `Vec` or `VecDeque` shouldn't really matter for the outsider.

Alternatively you could have turned vector into peekable iterator: https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.peekable I think peekable iterator will be even cleaner API that better describes what we're trying to do here.